spark/dl/src/main/resources/serialization/bigdl.proto

```diff
@@ -0,0 +1,110 @@
+syntax = "proto3";
+package serialization;
+message BigDLModule
+{
+    string name = 1; //module name
+    repeated BigDLModule subModules = 2; // sub modules
+    BigDLTensor weight = 3; // weight for each layer, serialized data are stored as either float or double
+    BigDLTensor bias = 4; // bias for each layer
+    repeated string preModules = 5; //pre modules of the same hierarchy
+    repeated string nextModules = 6; //next modules of the same hierachy
```

Comment:
Agree it's better to do this sanity check, but there is one exception for MapTable which replicates the module `n` times where n is the input table size. all the modules have the same reference thus threw exceptions, need to think about how to handle this situation...